gammapy/data/tests/test_obs_table.py

```diff
@@ -259,18 +259,26 @@ def test_select_time_box():
     obs_table_time = make_test_observation_table(
         n_obs=10,
         date_range=(datestart, dateend),
-        use_abs_time=True,
+        use_abs_time=False,
         random_state=random_state,
     )
 
     # test box selection in time: (time_start, time_stop) within value_range
     value_range = Time(["2012-01-01T01:00:00", "2012-01-01T02:00:00"])
     selection = dict(type="time_box", time_range=value_range)
     selected_obs_table = obs_table_time.select_observations(selection)
-    time_start = selected_obs_table["TSTART"]
-    assert (value_range[0] < time_start).all()
-    assert (time_start < value_range[1]).all()
+    time_start = selected_obs_table.time_start
+    time_stop = selected_obs_table.time_stop
 
+    assert (value_range[0] < time_start).all() & (time_stop < value_range[1]).all()
```

Comment:
Typical case of re-implementing the method in the test. Should be changed to a simple small test case and direct assert on output. But OK, that's old baggage that was already here.

Reply:
Yes that is not very good. But with random values it is difficult to do differently. Indeed the tests should be completely refactored.